kogito-usertasks-with-security-oidc-keycloak/README.md

```diff
@@ -0,0 +1,211 @@
+
+### Prerequisites
+ 
+You will need:
+  - Java 1.8.0+ installed 
+  - Environment variable JAVA_HOME set accordingly
+  - Maven 3.5.4+ installed
+  - jq tool
+  - Docker
+
+
+When using native image compilation, you will also need: 
+  - GraalVM 19.1+ installed
+  - Environment variable GRAALVM_HOME set accordingly
+  - Note that GraalVM native image compilation typically requires other packages (glibc-devel, zlib-devel and gcc) to be installed too, please refer to GraalVM installation documentation for more details.
+
+###Starting and Configuring the Keycloak Server
+
+To start a Keycloak Server you can use Docker and just run the following command:
+
+```
+docker run --name keycloak -e KEYCLOAK_USER=admin -e KEYCLOAK_PASSWORD=admin -p 8280:8080 quay.io/keycloak/keycloak
+```
+
+You should be able to access your Keycloak Server at [localhost:8280/auth](http://localhost:8280).
+
+Log in as the admin user to access the Keycloak Administration Console. Username should be admin and password admin.
+
+Import the realm configuration [file](config/quarkus-realm.json) to create a new realm. For more details, see the Keycloak documentation about how to 
+create a new realm.
+
+### Compile and Run in Local Dev Mode
+
+```
+mvn clean package quarkus:dev    
+```
+
+NOTE: With dev mode of Quarkus you can take advantage of hot reload for business assets like processes, rules, decision tables and java code. No need to redeploy or restart your running application.
+
+
+### Compile and Run using Local Native Image
+Note that this requires GRAALVM_HOME to point to a valid GraalVM installation
+
+```
+mvn clean package -Pnative
+```
+  
+To run the generated native executable, generated in `target/`, execute
+
+```
+./target/kogito-usertasks-with-security-oidc-keycloak-{version}-runner
+```
+
+### Use the application
+
+Examine OpenAPI via swagger UI at [http://localhost:8080/swagger-ui](http://localhost:8080/swagger-ui)
+
+
+### Submit a request to start new approval
+
+To make use of this application it is as simple as putting a sending request to `http://localhost:8080/approvals`  with following content 
+
+```
+{
+"traveller" : { 
+  "firstName" : "John", 
+  "lastName" : "Doe", 
+  "email" : "jon.doe@example.com", 
+  "nationality" : "American",
+  "address" : { 
+  	"street" : "main street", 
+  	"city" : "Boston", 
+  	"zipCode" : "10005", 
+  	"country" : "US" }
+  }
+}
+
+```
+
+Complete curl command can be found below:
+
+The application is using bearer token authorization and the first thing to do is obtain an access token from the Keycloak 
+Server in order to access the application resources. Obtain an access token for user john.
+
+```
+export access_token=$(\
+    curl -X POST http://localhost:8280/auth/realms/quarkus/protocol/openid-connect/token \
+    --user backend-service:secret \
+    -H 'content-type: application/x-www-form-urlencoded' \
+    -d 'username=john&password=john&grant_type=password' | jq --raw-output '.access_token' \
```

Comment:
this requires `jq` command to be present otherwise it will fail... as it was in my case :)

Reply:
@mswiderski ups, let me search for an alternative here :)